wpilibc/src/main/native/include/frc/util/Color.h

```diff
@@ -811,6 +812,23 @@ class Color {
     }
   }
 
+  /**
+   * Create a Color from a hex string. Throws an exception if the Hex String is
+   * invalid.
+   *
+   * @param hexString a string of the format <code>#RRGGBB</code>
+   * @return Color object from hex string.
+   */
+  static constexpr Color FromHexString(std::string hexString) {
+    if (hexString.length() != 7 || !hexString.starts_with("#")) {
+      throw std::invalid_argument("Invalid Hex String for Color");
+    }
+
+    int r, g, b;
+    std::sscanf(hexString.c_str(), "#%02x%02x%02x", &r, &g, &b);
```

Comment:
The return value of std::sscanf() isn't checked here. Furthermore, std::sscanf() isn't constexpr so it can't be used in this context.

Reply:
wpi/StringExtras.h has constexpr isHexDigit() and hexDigitValue() that could be used for this.